### PR TITLE
Homework 1 completed

### PR DESCRIPTION
**Pull request by daxdue**

### CODE REVIEW

File: src/test/java/ru/training/at/hw1/DataProviders.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+public interface DataProviders {
+    String SUM_LONG_DATA_PROVIDER = "sum_long_data_provider";
+    String SUM_DOUBLE_DATA_PROVER = "sum_double_data_provider";
+    String SUB_LONG_DATA_PROVIDER = "sub_long_data_provider";
+    String SUB_DOUBLE_DATA_PROVIDER = "sub_double_data_provider";
+    String MUL_LONG_DATA_PROVIDER = "mul_long_data_provider";
+    String MUL_DOUBLE_DATA_PROVIDER = "mul_double_data_provider";
+    String DIV_LONG_DATA_PROVIDER = "div_long_data_provider";
+    String DIV_DOUBLE_DATA_PROVIDER = "div_double_data_provider";
+    String DIV_BY_ZERO_DATA_PROVIDER = "div_by_zero_data_provider";
```
Comment (komyagin) [inline src/test/java/ru/training/at/hw1/DataProviders.java]: I see the names of the constants, but don't see appropriate keywords for a declaration.

File: src/test/java/ru/training/at/hw1/Tags.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package ru.training.at.hw1;
+
+public interface Tags {
+    String ADD_SUB = "addSub";
+    String MULT_DIV = "multDiv";
```
Comment (komyagin) [inline src/test/java/ru/training/at/hw1/Tags.java]: I see the names of the constants, but don't see appropriate keywords for a declaration.

File: src/test/java/ru/training/at/hw1/TestsDataProvider.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    @DataProvider(name = DataProviders.DIV_BY_ZERO_DATA_PROVIDER)
+    public Object[][] longDivByZeroDataProvider() {
+        return new Object[][] {{0, 0}, {454, 0}, {-1767, 0},
+            {Long.MAX_VALUE, 0}, {Long.MIN_VALUE, 0}};
+    }
+
+
+
```
Comment (komyagin) [inline src/test/java/ru/training/at/hw1/TestsDataProvider.java]: Why do you need so many blank lines?

File: src/test/java/ru/training/at/hw1/DataProviders.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package ru.training.at.hw1;
+
+public interface DataProviders {
```
Comment (komyagin) [inline src/test/java/ru/training/at/hw1/DataProviders.java]: Why do you use an interface for storing constants?

File: src/test/java/ru/training/at/hw1/Tags.java [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package ru.training.at.hw1;
+
+public interface Tags {
```
Comment (komyagin) [inline src/test/java/ru/training/at/hw1/Tags.java]: Why do you use an interface for storing constants?

File: pom.xml [hunk 6]
```diff
@@ -4,9 +4,109 @@
...
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>3.0.0-M5</version>
```
Comment (komyagin) [inline pom.xml]: If you use properties for testng and others why don't you use it for this?

File: pom.xml [hunk 7]
```diff
@@ -4,9 +4,109 @@
...
+        <testng.suite.file/>
+    </properties>
 
-</project>
\ No newline at end of file
+    <dependencies>
+        <dependency>
+            <groupId>com.epam</groupId>
+            <artifactId>calculator</artifactId>
+            <version>1.0</version>
```
Comment (komyagin) [inline pom.xml]: If you use properties for testng and others why don't you use it for this?

File: pom.xml [hunk 8]
```diff
@@ -4,9 +4,109 @@
...
+                <version>3.8.1</version>
+                <configuration>
+                    <source>${java.version}</source>
+                    <target>${java.version}</target>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-checkstyle-plugin</artifactId>
+                <version>3.1.2</version>
```
Comment (komyagin) [inline pom.xml]: If you use properties for testng and others why don't you use it for this?

**Review by komyagin — CHANGES_REQUESTED** [review]

File: src/test/java/ru/training/at/hw1/DataProviders.java (hunk 1 above)
Comment (daxdue) [reply src/test/java/ru/training/at/hw1/DataProviders.java]: Fixed.

File: src/test/java/ru/training/at/hw1/Tags.java (hunk 2 above)
Comment (daxdue) [reply src/test/java/ru/training/at/hw1/Tags.java]: Fixed.

File: src/test/java/ru/training/at/hw1/TestsDataProvider.java (hunk 3 above)
Comment (daxdue) [reply src/test/java/ru/training/at/hw1/TestsDataProvider.java]: Fixed.

**Review by komyagin — APPROVED** [review]

**Pull request merged**